src/main.cpp

```diff
@@ -1,9 +1,50 @@
 #include <cstdlib>
 #include <iostream>
+#include <boost/program_options.hpp>
+#include <iterator>
+
+#include "parse.h"
+#include "compiling.h"
+
+using namespace std;
```

Comment:
[same as above] Please don't import the entire std:: namespace. Prefix identifiers with std:: to make it explicit where they're coming from, or import only the part of the namespace you need (if possible).